codegen/value.go

```diff
@@ -39,14 +39,20 @@ type Register interface {
 
 type register struct {
 	Value
+	ctor func(iface interface{}) (Value, error)
 }
 
-func NewRegister() Register {
-	return &register{Value: ZeroValue()}
+func NewRegister(ctx context.Context) Register {
+	return &register{
+		Value: ZeroValue(ctx),
+		ctor: func(iface interface{}) (Value, error) {
+			return NewValue(ctx, iface)
```

Comment:
Preserving the context long-term like this feels a little icky. It might be better to expose a `NewValueWithPlatform` or add functional arguments to `NewValue` so we don't have to persist the context. But I think this is fine for now.

Reply:
yeah, I was trying to avoid changing the `(*Value).Set(interface{}))` signature, which is called in ~billion places.